salt/runners/venafiapi.py

```diff
@@ -54,311 +40,129 @@
+    key_password=None,
+    csr_path=None,
+    pkey_path=None,
+):
     '''
     Request a new certificate
 
-    Uses the following command:
-
-    .. code-block:: bash
-
-        VCert enroll -z <zone> -k <api key> -cn <domain name>
-
     CLI Example:
 
     .. code-block:: bash
 
         salt-run venafi.request <minion_id> <dns_name>
     '''
-    if password is not None:
-        if password.startswith('sdb://'):
-            password = __salt__['sdb.get'](password)
-
-    if zone_id is None:
-        zone_id = __opts__.get('venafi', {}).get('zone_id')
 
-    if zone_id is None and zone is not None:
-        zone_id = get_zone_id(zone)
-
-    if zone_id is None:
-        raise CommandExecutionError(
-            'Either a zone or a zone_id must be passed in or '
-            'configured in the master file. This id can be retreived using '
-            'venafi.show_company <domain>'
-        )
-
-    private_key = gen_key(minion_id, dns_name, zone, password)
-
-    csr = gen_csr(
-        minion_id,
-        dns_name,
-        zone=zone,
-        country=country,
-        state=state,
-        loc=loc,
-        org=org,
-        org_unit=org_unit,
-        password=password,
-    )
-
-    pdata = salt.utils.json.dumps({
-        'zoneId': zone_id,
-        'certificateSigningRequest': csr,
-    })
-
-    qdata = __utils__['http.query'](
-        '{0}/certificaterequests'.format(_base_url()),
-        method='POST',
-        data=pdata,
-        decode=True,
-        decode_type='json',
-        header_dict={
-            'tppl-api-key': _api_key(),
-            'Content-Type': 'application/json',
-        },
-    )
-
-    request_id = qdata['dict']['certificateRequests'][0]['id']
-    ret = {
-        'request_id': request_id,
-        'private_key': private_key,
-        'csr': csr,
-        'zone': zone,
-    }
+    log.info("Requesting Venafi certificate")
+    if zone is None:
+        log.error(msg=str("Missing zone parameter"))
+        sys.exit(1)
+
+    if key_password is not None:
+        if key_password.startswith('sdb://'):
+            key_password = __salt__['sdb.get'](key_password)
+    conn = _init_connection()
+
+    if csr_path is None:
+        request = CertificateRequest(common_name=dns_name, country=country, province=state, locality=loc,
+                                     organization=org, organizational_unit=org_unit, key_password=key_password)
+        zone_config = conn.read_zone_conf(zone)
+        log.info("Updating request from zone %s", zone_config)
+        request.update_from_zone_config(zone_config)
+    else:
+        log.info("Will use generated CSR from %s", csr_path)
+        log.info("Using CN %s", dns_name)
+        try:
+            with salt.utils.files.fopen(csr_path) as csr_file:
+                csr = csr_file.read()
+            request = CertificateRequest(csr=csr, common_name=dns_name)
+        except Exception as e:
+            log.error(msg=str(e))
+            sys.exit(1)
```

Comment:
The lint is complaining about this Exception catching. Looking at it, I don't see that any of the other runners follow this pattern at all.